be/src/formats/json/numeric_column.cpp

```diff
@@ -0,0 +1,150 @@
+                return Status::InvalidArgument(err_msg);
+            }
+        }
+
+        std::string err_msg = strings::Substitute("Unable to cast string value to BIGINT. value=$0, column=$1",
+                                                  std::string(sv.data(), sv.size()), name);
+        return Status::InvalidArgument(err_msg);
+    }
+}
+
+template <typename T>
+Status add_numeric_column(Column* column, const TypeDescriptor& type_desc, const std::string& name,
+                          simdjson::ondemand::value& value) {
+    auto numeric_column = down_cast<FixedLengthColumn<T>*>(column);
+
+    try {
+        simdjson::ondemand::json_type tp = value.type();
+        switch (tp) {
+        case simdjson::ondemand::json_type::number: {
+            return add_column_with_numeric_value(numeric_column, type_desc, name, value);
+        }
+
+        case simdjson::ondemand::json_type::string: {
+            return add_column_with_string_value(numeric_column, type_desc, name, value);
+        }
+
+        default: {
+            auto err_msg = strings::Substitute("Unsupported value type. Numeric type is required. column=$0", name);
+            return Status::InvalidArgument(err_msg);
+        }
+        }
+    } catch (simdjson::simdjson_error& e) {
+        auto err_msg = strings::Substitute("Failed to parse value as number, column=$0, error=$1", name,
+                                           simdjson::error_message(e.error()));
+        return Status::DataQualityError(err_msg);
+    }
+}
+
+template Status add_numeric_column<int128_t>(Column* column, const TypeDescriptor& type_desc, const std::string& name,
+                                             simdjson::ondemand::value& value);
+template Status add_numeric_column<int64_t>(Column* column, const TypeDescriptor& type_desc, const std::string& name,
+                                            simdjson::ondemand::value& value);
+template Status add_numeric_column<int32_t>(Column* column, const TypeDescriptor& type_desc, const std::string& name,
+                                            simdjson::ondemand::value& value);
+template Status add_numeric_column<int16_t>(Column* column, const TypeDescriptor& type_desc, const std::string& name,
+                                            simdjson::ondemand::value& value);
+template Status add_numeric_column<int8_t>(Column* column, const TypeDescriptor& type_desc, const std::string& name,
+                                           simdjson::ondemand::value& value);
+template Status add_numeric_column<double>(Column* column, const TypeDescriptor& type_desc, const std::string& name,
+                                           simdjson::ondemand::value& value);
+template Status add_numeric_column<float>(Column* column, const TypeDescriptor& type_desc, const std::string& name,
+                                          simdjson::ondemand::value& value);
```

Comment:
Pass mutable parameter by pointer, pass immutable parameter by const reference.

Reply:
Updated. The reference is replaced with pointer.